thunder/__init__.py

```diff
@@ -79,6 +79,13 @@
 pythonex = extend.get_executor("python")
 assert pythonex is not None
 
+# Avoids the allocator thrashing issue in PyTorch NCCL backend.
+# This is a workaround for the problem described in
+# https://github.com/Lightning-AI/lightning-thunder/issues/420
+# Unfortunately, PyTorch reads this value only once at the ProcessGroupNCCL initialization time,
+# so we need to set it before the first call to ProcessGroupNCCL.
+# Hopefully, Thunder is imported before any other PyTorch code that uses NCCL.
```

Comment:
Is there any way for us to detect that `ProcessGroupNCCL` has been initialized already? This way we can produce a good warning.